apps/voting/app/src/math-utils.js

```diff
@@ -61,44 +63,99 @@ export function safeDiv(num, denom) {
   return denom ? num / denom : 0
 }
 
-// Get a list of rounded 0 => `total` numbers, from a list of 0 => 1 values.
-// If the total of the values is exactly 1, the total of the resulting values
-// will be exactly `total`.
-export function scaleValuesSet(values, digits = 0, total = 100) {
-  const digitsMultiplicator = Math.pow(10, digits)
+// Scale to `total` a set of values summing to 1.
+// Note that the accuracy of `values` is constrained to a set amount of decimals, as BN.js doesn't
+// support decimal operations
+export function scaleBNValuesSet(
+  values = [],
+  total = new BN(100),
+  correctionLimit = 0.001
```

Comment:
What do you think about keeping the inputs and outputs to a single number type (normal JS numbers)? Mixing them up is invariably going to lead to a really bad time at some point or another, and for the purposes of this function, I think it's enough to completely abstract away the BN usage.

Reply:
Makes sense, there is a tradeoff:

![image](https://user-images.githubusercontent.com/3072458/56801490-d5b10800-681d-11e9-8bcb-a55a4a1805a4.png)

Reply:
Hmm, just checked what we need this for and indeed sometimes we do use this to scale balances (e.g. creating a distribution over multiple vote options), so we do need to expose `total` as a BN.

Let's backtrack a bit... perhaps we can do the `.toNumber()` in `percentageList()` (since that should never use a different `total` than 100)?

Reply:
Better approach yes, thanks for the suggestion. Done.